### ISSUE TITLE
Client 3.4.1 Syncing does not work properly.

### ISSUE THREAD

**Issue by Ithu123** [nextcloud/desktop#4141]
### Expected behaviour
Files get synced, while displaying a blue icon on the file. Afterwards file is synced with server and displaying a green Icon.
Using Windows Desktop client 3.3.6, Explorer this worked.

### Actual behaviour
Client starts syncing, displays blue icon, then stops and ignores the file completly. Client also says everthing is synced.
![grafik](https://user-images.githubusercontent.com/65660176/148267542-4b6f0102-2946-4994-89c0-226eb000aee8.png)
![grafik](https://user-images.githubusercontent.com/65660176/148267912-4352c8cd-b419-4c08-9ac8-38c6888e7021.png)

### Steps to reproduce
1. Update from 3.6.1 to 3.4.1
2. Create new file

### Client configuration
Client version: 3.4.0 and 3.4.1

Operating system: Windows 10

OS language: German

Installation path of client: C:\Program Files\Nextcloud

### Server configuration

Dockerized Nextcloud 23.0.0, with sshfs Storage

Nextcloud version: 23

### Logs
I would provide logs, but there seem to be non for the faild sync.

**Comment by ttyS0** [comment]
I experienced this on macOS as well, and rolled the desktop client back to `3.3.6` to have consistent syncing. There didn't seem to be any rhyme or reason to the files/folders that sync skipped. 

I'm also using sever version `v23.0.0`.

**Comment by araschke** [comment]
I can confirm the behaviour described by lthu123. Same configuration (Windows 10, Desktop-Client 3.4.1, Server-Version 23.0)

**Comment by TimGitHoub** [comment]
Same problem here and same configuration.

**Comment by koppd** [comment]
What's the file size of your test files? 
Zero? Then it's the same issue as in bug #4157.

**Comment by ttyS0** [comment]
In my case I don't think that (sync blocked by zero length files) was a factor. _Some_ files would sync, and others would not. The files that did not sync were not zero length files. For me, this was apparently an issue for a lot longer than I realized, as when I reverted to desktop client `v3.3.6` there were a _lot_ of files that were suddenly synced from my local system to the server of varying ages, sizes, locations.

**Comment by Ithu123** [comment]
> What's the file size of your test files? Zero? Then it's the same issue as in bug #4157.

Nah, at least with my testing on Client v.3.4.0 file sizes varied from few kB to few MB. File size did not matter, as far as i can tell. In this case particular case file size varied in the kB range.

**Comment by JorgeGNL** [comment]
I have the same issue with v3.4.1 (Windows 11)

v3.3.6 does not have this problem, I recommend downgrading:
https://github.com/nextcloud/desktop/releases/tag/v3.3.6

**Comment by koukjo** [comment]
The same issue appears in Archlinux nextcloud client v3.4.1  (Nextcloud v23.0.0)
When copying a file in the local folder nextcloud client syncs but no files has been uploaded in the server. 
Downgrading to v3.3.6 also worked for me.

**Comment by sangfroid70** [comment]
Same issue since I upgraded to Nextcloud Server 23.0.0
Client : 
```
Nextcloud version 3.4.1git
Git revision fa32c100148925092c9a320e6f8e2e0130b9cb6a
Using Qt 5.15.2, built against Qt 5.15.2
Using Qt platform plugin 'xcb'
Using 'OpenSSL 1.1.1m  14 Dec 2021'
Running on Arch Linux, x86_64
```
How to reproduce : 
put a file in a folder of my laptop, the client seems starting to sync and client doesn't respond anymore.
No error or warning in logs.

**Comment by Lars1963** [comment]
After reading this issue, I realized, that I have the problem with client 3.4.1 on windows10 and nc 23.0.1 RC1. After downgrading to client 3.3.6 over hundreds of new vacation-fotos were then synced between PC and server (and then with the androidclient on several other devices). 

I'm still trying to figur out, if the syncing between android-devices and server is also involved (which would mean this is really a server-problem). And I'm not sure if this has led to data-loss; still investigating that.

**Comment by pnwwebworks** [comment]
I have the same issue, Windows client 3.4.1, Windows 10 x64 21H2, NextCloud 23.0.0

**Comment by KimmoJ** [comment]
I'm seeing this also, 3.4.1 on Windows 10 vs a freshly installed NC 23. Something is definitely off. I have partial folders uploading and others not, only some files are tagged as synced in the Windows Explorer, it's like the client can't "see them" or whatever. I connected to the NC via Webdav and Windows Explorer and  uploaded a folder that way and the client then downloaded the files from the NC, but things I add in the client folder for upload gets partially uploaded only. Very disturbing behavior, I wish it would just fail so you know, now I'm just not sure what's up there and what isn't. Something with the client and/or NC 23 "bulk upload" improvements...?

**Comment by Emporea** [comment]
Anyone has infos about this problem occuring on Nextcloud Server <23?

**Comment by holzfelix** [comment]
I have the Same issue!

**Comment by holzfelix** [comment]
I can confirm that a downgrade to 3.3.6 solves the problem. Server is V23

The 3.4 client is broken.

**Comment by JMLatGitHub** [comment]
> Anyone has infos about this problem occuring on Nextcloud Server <23?

Hi there, @Emporea.

My setup:
Debian Sid (Testing V12, Bookworm), Linux-Kernel SMP Debian 5.15.15-1 (2022-01-18) x86_64 GNU/Linux); Redis Server 6.0.16; MariaDB 10.5.12; STUN/TURN Server (coTURN) 4.5.2 'dan Eider'; PHP 8.0.14; NGINX 1.21.6; Nextcloud Server 21.0.2.1

Testing constellations:
Created/copied four directories with 12 files each into four different Sync Folder connections, one of them defined as Virtual Files connection.

Client scenarios:
Nextcloud Desktop Client 3.3.6: Sync successful.
Nextcloud Desktop Client 3.4.1: Sync successful.
Nextcloud Desktop Client 3.4.2: Sync successful, but without Windows File Manager overlay icons that should indicate the sync state of folders and files (bug reported already [#4223](https://github.com/nextcloud/desktop/issues/4223)). I have verified the successful synchronization via web access to the server.

**Comment by JMLatGitHub** [comment]
Well, in the meantime, I basically assume with every new announcement/major change that there will be correspondingly big problems... that's why I'm currently still at V21 on the server side. 🤷‍♂️

**Comment by DmitriiOdintcov** [comment]
I'm experiencing the same problem with 3.4.2 on Windows with the exception that sometimes the folder itself might appear synched (green checkmark), but the underlying files still aren't.

I noticed that once some sort of change is introduced, for example one of those files is renamed, Nextcloud does detect and sync it - now together with everything else in the directory.

Perhaps this is the result of an oversight in some attempt at optimising how sync traverses directory trees?..

**Comment by Emporea** [comment]
> Well, in the meantime, I basically assume with every new announcement/major change that there will be correspondingly big problems... that's why I'm currently still at V21 on the server side. 🤷‍♂️

Yeah I am on 22. Besides the [timestamp to epoch 0 problem ](https://github.com/nextcloud/desktop/issues/4016) its working so far I guess. But who knows.

**Comment by JMLatGitHub** [comment]
The reset of the timestamps is a real disaster, as well as the unreliability/inconsistency of the file synchronization. Until these two major bugs are solved, I will not update to version >21. How tricky and problematic the whole thing is, you can probably guess from the now one month delay of the release of the maintenance version 23.0.1...

**Comment by Korsani** [comment]
> I can confirm that a downgrade to 3.3.6 solves the problem. Server is V23
> 
> The 3.4 client is broken.

Same issue here : 
- MacOS 11.6
- NC Server 23.0
Client take 100% cpu and get stuck upon syncing TO the server. It seems to sync well FROM the server.
And forcing a sync via nextcloudcmd clean the stuff (allows the client to not eat all CPU)
Downgrading to 3.3.6 seems to fix.

**Comment by neumantm** [comment]
Can confirm that this issue doesn't seem to appear on older versions of the server (21 in my case).

**Comment by DmitriiOdintcov** [comment]
> Can confirm that this issue doesn't seem to appear on older versions of the server (21 in my case).

Isn't this a client issue?

**Comment by m707** [comment]
I am afraid it's a combined issue :(. 3.4.1 seems to be working with old server (<22) where 3.3.6 seems to work with everything.

Am 30. Januar 2022 17:37:31 MEZ schrieb Dmitrii Odintcov ***@***.***>:
>> Can confirm that this issue doesn't seem to appear on older versions of the server (21 in my case).
>
>Isn't this a client issue?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/nextcloud/desktop/issues/4141#issuecomment-1025180948
>You are receiving this because you are subscribed to this thread.
>
>Message ID: ***@***.***>
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by DmitriiOdintcov** [comment]
> I am afraid it's a combined issue :(. 3.4.1 seems to be working with old server (<22) where 3.3.6 seems to work with everything.

Schade

**Comment by m707** [comment]
After update to 3.4.2 my issues are fixed again.

Glücklich ;)

**Comment by holzfelix** [comment]
> After update to 3.4.2 my issues are fixed again.
> 
> Glücklich ;)

I can't confirm that

**Comment by Ralf-J** [comment]
Unfortunately can also not confirm that the issue is resolved with 3.4.2. Using it on MacOS 12; server Version is 23.

**Comment by pnwwebworks** [comment]
I seem to be having the issue even more now. I have 3.3.6 installed and was working great on a computer that has had NextCloud desktop client for at least a year. The server version has upgraded through time, while this older PC was using it. Ran into the 3.4.X problem and commented above.

Now I have a new PC, so I installed 3.3.6 directly to avoid the problem, but it did not resolve the issue. The new PC does not detect new files unless the client is closed and reopened, only randomly in random folders. Copied settings and data from old PC, but the issue still exists. Tried upgrading to 3.4.2 and down to 3.3.3, the same problem exists. 

The server version is still 23.0.0.10.

I have also tried files in varying sizes, from a few KB to many MB. No change. If the file exists before, and I delete it, it detects the delete. If I re-add, it sometimes detects the re-added files. New files get ignored entirely.

Any tips to get it at least semi-working? I am about ready to blow my server away and go back to 22.X, just a lot of time and work that I want to avoid if possible.

**Comment by Emporea** [comment]
The changelog isnt on the website yet but you could atleast try 23.0.1 or even 23.0.2rc

**Comment by lars-becker** [comment]
Whatever it is, is has something to do with the new bulk upload. The last line of the client log always look similar to this:

```
[ OCC::CookieJar::cookiesForUrl ]:	QUrl("https://***.de/remote.php/dav/bulk") requests: (QNetworkCookie("__Host-nc_sameSiteCookielax=true; secure; HttpOnly; SameSite=lax; expires=Fri, 31-Dec-2100 23:59:59 GMT; domain=***.de; path=/"), QNetworkCookie("__Host-nc_sameSiteCookiestrict=true; secure; HttpOnly; SameSite=strict; expires=Fri, 31-Dec-2100 23:59:59 GMT; domain=***.de; path=/"), QNetworkCookie("oc_sessionPassphrase=***; secure; HttpOnly; SameSite=Lax; domain=***.de; path=/"), QNetworkCookie("***; secure; HttpOnly; SameSite=Lax; domain=cloud.europa-union-hamburg.de; path=/"))
2022-02-16 01:06:32:031 [ info nextcloud.sync.networkjob /build/nextcloud-client/src/nextcloud-client/src/libsync/abstractnetworkjob.cpp:361 ]:	OCC::PutMultiFileJob created for "***.de" + "" "OCC::BulkPropagatorJob"
```

Is there a possibility to disable bulk uploads? This feature is highly disruptive because it's impossible to work with the sync client.

I don'T think that it has anything to do with the virtual files, because I'm using the Linux version without virtual files and it happens on my desktop as well as on those who are using windows with and without virtual files.

**Comment by LecrisUT** [comment]
So is it confirmed that with 23.0.2 and with bulk uploads disabled, this issue is not occurring. I would not call it resolved, but at the very least, if this issue is directly linked to #4243, then we should close this one and move further discussions there.

Particularly can we get some updated feedback from:
- [ ] @juliushaertl or anyone doing a similar test about the file size mattering
- [ ] @Ithu123 about the file size not mattering
- [ ] @JorgeGNL is it related to Federation share (might be better to create a separate issue there)

Anybody else if they can give a detailed test that they have done particularly noting:
- The versions of the Nextcloud server (particularly if it's 23.0.2), bulk upload setting and other v2 sync features (probably I have the name of the new sync engine wrong)
- What configurations work or not with the latest client v3.4.2.
- Setup of the files that create sync issues
- Any other notable settings or conditions

**Comment by juliushaertl** [comment]
Yes, for me the issue was mitigated with the bulk upload disabling, further more there was https://github.com/nextcloud/desktop/pull/4263 for additional hardening of the sync process. 

In regards to the bulk upload topic there is also https://github.com/nextcloud/server/issues/29702 for further tracking of the server side issues.

**Comment by Ithu123** [comment]
So i finally had the time to update and it seems, that with bulk upload disabled the problem is gone. Unfortunately i am now witnessing other Problems.

**Comment by maja68** [comment]
Hi @Ithu123,

do you mind to give a reference to the issues you are facing? I mean, which other problems you are witnessing?

**Comment by Maarrk** [comment]
> Anybody else if they can give a detailed test that they have done particularly noting:

Had the described issue with this setup:

- Server: 23.0.0
- Client: Windows 3.4.1
- Everything with default settings
- Files were just put into the synchronised folder in bulk
    - I don't see anything specific with the files that didn't synchronise
    - I think it didn't happen to any file that I put there on it's own (it was only in the big backup folders)
    - There were circa 7400 files that didn't sync out of 17000, ( ca. 1GB out of 51GB)
- Connection over HTTPS, AFAIK no E2EE

Steps done to fix:

- [Updated via command line](https://docs.nextcloud.com/server/latest/admin_manual/maintenance/update.html#using-the-command-line-based-updater)

Results:

- Server is now 23.0.3
- Missing files synchronised in a few minutes
- Didn't see any issues yet, done some changes through client and web, things seem to work normally

**Comment by Quadratball** [comment]
> > Anybody else if they can give a detailed test that they have done particularly noting:
> 
> Had the described issue with this setup:
> 
>     * Server: 23.0.0
> 
>     * Client: Windows 3.4.1
> 
>     * Everything with default settings
> 
>     * Files were just put into the synchronised folder in bulk
>       
>       * I don't see anything specific with the files that didn't synchronise
>       * I think it didn't happen to any file that I put there on it's own (it was only in the big backup folders)
>       * There were circa 7400 files that didn't sync out of 17000, ( ca. 1GB out of 51GB)
> 
>     * Connection over HTTPS, AFAIK no E2EE
> 
> 
> Steps done to fix:
> 
>     * [Updated via command line](https://docs.nextcloud.com/server/latest/admin_manual/maintenance/update.html#using-the-command-line-based-updater)
> 
> 
> Results:
> 
>     * Server is now 23.0.3
> 
>     * Missing files synchronised in a few minutes
> 
>     * Didn't see any issues yet, done some changes through client and web, things seem to work normally

Thanks, was running 23.0.0 and updated to 23.0.4 and everything works like charm.

**Comment by LecrisUT** [comment]
I believe  we should close this issue for now as the title and most of the discussion are irrelevant now. Still keep in mind that the bulk sync with small/empty files is not yet tested.
